nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -848,6 +849,20 @@ ngx_http_modsecurity_body_handler(ngx_http_request_t *r)
     ngx_http_core_run_phases(r);
 }
 
+static void
+ngx_http_calculate_waf_latency(ngx_http_request_t *r, ngx_time_t *start_time)
+{
+    ngx_time_t *end_time;
+    ngx_msec_int_t   ms;
+    ngx_time_update();
+    end_time = ngx_timeofday();
+
+    ms = (ngx_msec_int_t)
+        ((end_time->sec - start_time->sec) * 1000 + (end_time->msec - start_time->msec));
+    ms = ngx_max(ms, 0);
+    r->waf_latency = ms;
```

Comment:
As Paresh pointed out in the nginx PR -- we shouldn't be modifying nginx structure, as nginx should not know anything about WAF. Instead, I wonder if it's possible to use the [Configuration Variables API](https://www.nginx.com/resources/wiki/extending/api/variables/) to expose `waf_latency` as a configuration variable?

I'm not entirely sure of how this works or what many of the terminologies mean. I'm suggesting this because I noticed this is how the `ngx_http_upstream_module` exposes its embedded variables (_e.g._ `$upstream_addr`, `$upstream_status`, `$upstream_response_time`, _etc_) to the nginx config, which we used to [customize](https://msazure.visualstudio.com/One/_git/Networking-AppGW?path=%2Fsrc%2FTenant%2FApplicationGateway%2FRuntimeComponents%2FNginxApplicationGatewayConfigTransformer.cs&version=GBdevelop&line=1101&lineEnd=1103&lineStartColumn=1&lineEndColumn=93&lineStyle=plain&_a=contents) the log format the way you currently have, without intruding the definition of `ngx_http_request_s` in nginx core.

Reply:
Modified the variable locally and not using the ngx_http_request_s  struct anymore